### ISSUE TITLE
[Hosted Che] update to 7.17.0 is failing due to changes in Openshift4TrustedCAProvisioner

### ISSUE THREAD

**Issue by ibuziuk** [eclipse-che/che#17576]
### Describe the bug
Provision custom CA certificates into every container of the workspace pod have been implemented as part of:
- https://github.com/eclipse/che/issues/17407
- https://github.com/eclipse/che/pull/17460

Currently, the changes prevent Hosted Che update to 7.17.0 upstream version. Basically, with the current implementation, it is not possible to opt-out from the usage of the `Openshift4TrustedCAProvisioner` (cm check should not happen for Hosted Che case since SA has no permissions to do it). In general, there should be a possibility to disable custom CA certificates into every container of the workspace since this case is addressing a particular case with Eclipse Che deployed behind the proxy.)

### Che version
7.17.0

### Expected behavior
Successful Hosted Che deployment against the latest 7.17.0 upstream
for Hosted Che case Openshift4TrustedCAProvisioner should not be called - https://github.com/eclipse/che/pull/17460/files#diff-1608a763bf13a812bb359246b8a62d96R52

### Runtime
   - [x] Openshift (include output of `oc version`)

### Environment
   - [x] OSD v4 che-dev cluster

### Eclipse Che Logs
```
2020-08-04 15:27:47,167[ost-startStop-1]  [ERROR] [o.a.c.c.C.[.[localhost].[/api] 175]  - Exception sending context initialized event to listener instance of class [org.eclipse.che.inject.CheBootstrap]
com.google.inject.CreationException: Unable to create injector, see the following errors:
1) Error injecting constructor, io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://172.30.0.1/api/v1/namespaces/prcheck-1959/configmaps?labelSelector=config.openshift.io%2Finject-trusted-cabundle%3Dtrue. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. configmaps is forbidden: User "system:serviceaccount:prcheck-1959:rhche" cannot list resource "configmaps" in API group "" in the namespace "prcheck-1959".
  at org.eclipse.che.workspace.infrastructure.openshift.provision.Openshift4TrustedCAProvisioner.<init>(Openshift4TrustedCAProvisioner.java:58)
  at org.eclipse.che.workspace.infrastructure.openshift.provision.Openshift4TrustedCAProvisioner.class(Openshift4TrustedCAProvisioner.java:58)
  while locating org.eclipse.che.workspace.infrastructure.openshift.provision.Openshift4TrustedCAProvisioner
1 error
	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:543)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:186)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
	at com.google.inject.Guice.createInjector(Guice.java:87)
	at org.everrest.guice.servlet.EverrestGuiceContextListener.getInjector(EverrestGuiceContextListener.java:141)
	at com.google.inject.servlet.GuiceServletContextListener.contextInitialized(GuiceServletContextListener.java:45)
	at org.everrest.guice.servlet.EverrestGuiceContextListener.contextInitialized(EverrestGuiceContextListener.java:86)
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4689)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5155)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:743)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:719)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
	at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:970)
	at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:1840)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://172.30.0.1/api/v1/namespaces/prcheck-1959/configmaps?labelSelector=config.openshift.io%2Finject-trusted-cabundle%3Dtrue. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. configmaps is forbidden: User "system:serviceaccount:prcheck-1959:rhche" cannot list resource "configmaps" in API group "" in the namespace "prcheck-1959".
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:568)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:505)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:471)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:430)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:412)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(BaseOperation.java:151)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:621)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:70)
	at org.eclipse.che.workspace.infrastructure.openshift.provision.Openshift4TrustedCAProvisioner.<init>(Openshift4TrustedCAProvisioner.java:68)
	at org.eclipse.che.workspace.infrastructure.openshift.provision.Openshift4TrustedCAProvisioner$$FastClassByGuice$$afa88f6b.newInstance(<generated>)
	at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
	at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:211)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:182)
	... 18 common frames omitted
04-Aug-2020 15:27:47.169 SEVERE [localhost-startStop-1] org.apache.catalina.core.StandardContext.startInternal One or more listeners failed to start. Full details will be found in the appropriate container log file
04-Aug-2020 15:27:47.169 SEVERE [localhost-startStop-1] org.apache.catalina.core.StandardContext.startInternal Context [/api] startup failed due to previous errors
```
### Additional context
Hosted Che PR check https://github.com/redhat-developer/rh-che/pull/1950

**Comment by tolusha** [comment]
@mshaposhnik 
Simple check if `CHE_TRUSTED__CA__BUNDLES__CONFIGMAP' env is set.

**Comment by ibuziuk** [comment]
@mshaposhnik if the fix is simple enough would be awesome to have in 7.17.0, otherwise we could wait for 7.17.1 next week

**Comment by tolusha** [comment]
Some more details.
`che-operator` detects if proxy is configured. If so then it creates the `ca-certs` configmap and allows OpenShift Network operator to inject ca-bundle. In additional it adds the key `CHE_TRUSTED__CA__BUNDLES__CONFIGMAP` with `ca-certs` value into `che` configmap.

**Comment by ibuziuk** [comment]
thanks for clarification but in Hosted Che case operator is not used for deployment, also in general we are runnning against 3.11 infra and the fact that `Openshift4TrustedCAProvisioner` is called is not expected

**Comment by ibuziuk** [comment]
The fix was cherrypicked to 7.17.x and should be available on Hosted Che with the 7.17.1 release this week cc: @nickboldt @mkuznyetsov
